docs/contributing/provider-design.md

```diff
@@ -0,0 +1,127 @@
+
+## Resource Type Considerations
+
+The general heuristic of whether or not to create a separate Terraform resource is if the AWS service API implements create, read, and delete operations for a component. Terraform resources work best as the smallest blocks of infrastructure, where configurations can build abstractions on top of these such as [Terraform Modules](https://www.terraform.io/docs/modules/). However, not all AWS service API functionality falls cleanly into components to be managed with those types of operations. It may be necessary to consider other patterns of mapping API operations to Terraform resources and operations. This section highlights these design recommendations, such as when to consider implemantations within the same Terraform resource or as a separate Terraform resource.
+
+Please note: the overall design and implementation across all AWS functionality is federated. This means that individual services may implement concepts and use terminology differently. As such, this guide may not be fully exhaustive in all situations. The goal is to provide enough hints about these concepts and towards specific terminology to point contributors in the correct direction, especially when researching prior implementations.
+
+### Authorization and Acceptance Resources
+
+Certain AWS services use an authorization-acceptance model for cross-account associations or access. Some examples of these can be found with:
+
+* Direct Connect Association Proposals
+* GuardDuty Member Invitations
+* RAM Resource Share Associations
+* Route 53 VPC Associations
+* Security Hub Member Invitations
+
+This type of API model either implements a form of an invitation/proposal identifier that can be used to accept the authorization, otherwise it may just be a matter of providing the desired AWS Account identifier as the target of an authorization. In the latter case, the acceptance is implicit when creating the other half of the association.
+
+To model the separate invitation/proposal in Terraform resources:
+
+* Typically the authorization side has separate and sufficient creation and read API functionality to create an "invitation" or "proposal" resource.
+* If the acceptance side has separate and sufficient accept, read, and reject API functionality, then an "accepter" resource may be created, where the operations are mapped as:
+    * Create: Accepts the invitation/proposal.
+    * Read: Reads the invitation/proposal to determine status. If not found, then it should fallback to reading the API resource associated with the invitation/proposal. As evidenced in some previous API implementations, the invitation/proposal may be temporary and removed from the API after an indeterminate amount of time which may not be documented or easily discoverable from testing.
+    * Delete: Rejects or otherwise deletes the invitation/proposal.
```

Comment:
```suggestion
* Follow the naming in the AWS service API to determine whether to use the term "invitation" or "proposal."
* For the originating account, create an "invitation" or "proposal" resource. Make sure that the AWS service API has operations for creating and reading invitations. 
* For the responding account, create an "accepter" resource. Ensure that the API has operations for accepting, reading, and rejecting invitations in the responding account. Map the operations as follows:
	* Create: Accepts the invitation.
	* Read: Reads the invitation to determine its status. Note that in some APIs, invitations expire and disappear, complicating associations. If a resource does not find an invitation, the developer should implement a fall back to read the API resource associated with the invitation/proposal.
	* Delete: Rejects or otherwise deletes the invitation.
```